src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc

```diff
@@ -148,168 +170,184 @@ const std::string LightName(const smart_objects::SmartObject& light_name) {
   return ok ? name : "unknown";
 }
 
-bool CheckLightDataByCapabilities(
+mobile_apis::Result::eType GetItemCapability(
     const smart_objects::SmartObject& capabilities,
-    const smart_objects::SmartObject& light_data) {
+    const std::map<std::string, std::string>& mapping,
+    const std::string& request_parameter,
+    const mobile_apis::Result::eType& switched_off_result) {
+
+  const auto it = mapping.find(request_parameter);
+
+  if (it == mapping.end()) {
+    LOG4CXX_DEBUG(logger_,
+                "Parameter "
+                    << request_parameter
+                    << " doesn't exist in capabilities.");
+    return mobile_apis::Result::UNSUPPORTED_RESOURCE;
+  }
+
+  const std::string& caps_key = it->second;
+  
+  LOG4CXX_DEBUG(logger_,
+                "Checking request parameter "
+                    << request_parameter
+                    << " with capabilities. Appropriate key is " << caps_key);
+
+  if (!capabilities.keyExists(caps_key)) {
+    LOG4CXX_DEBUG(logger_,
+                  "Capability " << caps_key
+                                << " is missed in RemoteControl capabilities");
+    return mobile_apis::Result::UNSUPPORTED_RESOURCE;
+  }
+
+  if (!capabilities[caps_key].asBool()) {
+    LOG4CXX_DEBUG(logger_,
+                  "Capability "
+                      << caps_key
+                      << " is switched off in RemoteControl capabilities");
+    return switched_off_result;
+  }
+
+  return mobile_apis::Result::SUCCESS;
+}
+
+mobile_apis::Result::eType GetLightDataCapabilities(
+    const smart_objects::SmartObject& capabilities,
+    const smart_objects::SmartObject& control_data) {
   LOG4CXX_AUTO_TRACE(logger_);
-  std::map<std::string, std::string> lightCapsMapping = {
-      {message_params::kId, strings::kName},
-      {message_params::kDensity, strings::kDensityAvailable},
-      {message_params::kSRGBColor, strings::kSRGBColorSpaceAvailable}};
-  auto it = light_data.map_begin();
-  for (; it != light_data.map_end(); ++it) {
-    if (message_params::kStatus == it->first ||
-        message_params::kId == it->first) {
+  std::map<std::string, std::string> mapping =
+      GetLightCapabilitiesMapping();
+
+  for (auto it = control_data.map_begin(); it != control_data.map_end(); ++it) {
```

Comment:
@KVGrygoriev Can't we use range-based loop here? 
`for (auto&& var : sequence)`